### ISSUE TITLE
LdapUserInfo  Avatar picture broken

### ISSUE THREAD

**Issue by PhilippSelenium** [buildbot/buildbot#5793]
Since I cannot find any reference to my custom code I assume it is an error in buildbot?

```
	Traceback (most recent call last):
	  File "python3.8/site-packages/buildbot/www/avatar.py", line 79, in render_GET
	    return self.asyncRenderHelper(request, self.renderAvatar)
	  File "python3.8/site-packages/buildbot/www/resource.py", line 72, in asyncRenderHelper
	    d = _callable(request)
	  File "python3.8/site-packages/twisted/internet/defer.py", line 1613, in unwindGenerator
	    return _cancellableInlineCallbacks(gen)
	  File "python3.8/site-packages/twisted/internet/defer.py", line 1529, in _cancellableInlineCallbacks
	    _inlineCallbacks(None, g, status)
	--- <exception caught here> ---
	  File "python3.8/site-packages/twisted/internet/defer.py", line 654, in _runCallbacks
	    current.result = callback(current.result, *args, **kw)
	  File "python3.8/site-packages/buildbot/www/resource.py", line 96, in failHttpError
	    f.trap(Error)
	  File "python3.8/site-packages/twisted/python/failure.py", line 460, in trap
	    self.raiseException()
	  File "python3.8/site-packages/twisted/python/failure.py", line 488, in raiseException
	    raise self.value.with_traceback(self.tb)
	  File "python3.8/site-packages/twisted/internet/defer.py", line 654, in _runCallbacks
	    current.result = callback(current.result, *args, **kw)
	  File "python3.8/site-packages/buildbot/www/resource.py", line 89, in failHttpRedirect
	    f.trap(Redirect)
	  File "python3.8/site-packages/twisted/python/failure.py", line 460, in trap
	    self.raiseException()
	  File "python3.8/site-packages/twisted/python/failure.py", line 488, in raiseException
	    raise self.value.with_traceback(self.tb)
	  File "python3.8/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
	    result = g.send(result)
	  File "python3.8/site-packages/buildbot/www/avatar.py", line 91, in renderAvatar
	    res = yield method.getUserAvatar(email, username, size, self.defaultAvatarFullUrl)
	builtins.TypeError: getUserAvatar() takes 4 positional arguments but 5 were given
```

Buildbot master version: 2.10.1
buildbot worker version: 2020.08.31
Python version: 3.8.5

avatar plugin: buildbot.plugins.util.LdapUserInfo

**Comment by tardyp** [comment]
Helo,

Can you tell us which avatar plugin are you using?

**Comment by PhilippSelenium** [comment]
buildbot.plugins.util.LdapUserInfo

**Comment by tardyp** [comment]
There has been a PR recently which improved the avatar api
https://github.com/buildbot/buildbot/pull/5753

it looks like the LDAPUserInfo one was not updated.

Can you have a look at it, it is probably just adding the additional parameter to the method. (and ignore it)

**Comment by PhilippSelenium** [comment]
No, because I did not break it.

**Comment by tardyp** [comment]
@PhilippSelenium I a sorry about this feedback...

We rely on the community to maintain features they care about.

**Comment by PhilippSelenium** [comment]
I'm sorry too, but I just do not have the time and it does not yet bug me enough to fix it.

**Comment by lephilousophe** [comment]
Hi there,
I don't know why but I got the notification only today.

Sorry I overlooked the LDAPUserInfo.
I will look at it this evening to fix it.

**Comment by lephilousophe** [comment]
For the record, after a little investigation problem was caused by #5420.
My PR hasn't landed in 2.10 branch.
Will submit a fix PR anyway.
